satellite/analytics/service.go

```diff
@@ -175,9 +198,27 @@ func (service *Service) TrackEvent(eventName string, userID uuid.UUID) {
 		service.log.Error("Invalid client-triggered event", zap.String("eventName", eventName))
 		return
 	}
-
 	service.enqueueMessage(segment.Track{
 		UserId: userID.String(),
 		Event:  eventName,
 	})
 }
+
+// TrackLinkEvent sends an arbitrary event and link associated with user ID to Segment.
+// It is used for tracking occurrences of client-side events.
+func (service *Service) TrackLinkEvent(eventName string, userID uuid.UUID, link string) {
+	// do not track if the event name is an invalid client-side event
+
+	props := segment.NewProperties()
```

Comment:
nit - move these two lines below the `if !service.clientEvents[eventName] {` block.